lxd/instance/drivers/driver_qemu.go

```diff
@@ -1903,25 +1910,35 @@ func (d *qemu) deviceStop(dev device.Device, instanceRunning bool) error {
 		return err
 	}
 
-	if instanceRunning {
-		// Detach NIC from running instance.
-		if configCopy["type"] == "nic" {
-			err = d.deviceDetachNIC(dev.Name())
-			if err != nil {
-				return err
+	if runConf != nil {
```

Comment:
Before `d.deviceDetachNIC` and `deviceDetachBlockDevice` were called even without `runConf`, how come this has changed?